src/HttpApplicationRunner.php

```diff
@@ -208,4 +226,93 @@ private function registerErrorHandler(ErrorHandler $registered, ?ErrorHandler $u
 
         $registered->register();
     }
+
+    private function removeBodyByStatusMiddleware(ResponseInterface $response): ResponseInterface
```

Comment:
Initially a separate method to send headers was for optimization purposes, i.e. not to actually generate and the discard the body.